software/contrib/consequencer.py

```diff
@@ -47,6 +47,22 @@ def __init__(self):
         self.SN=p.SN
         self.HH=p.HH
 
+        # Initialize sequencer pattern probabiltiies
+        self.BdProb = p.BdProb
+        self.SnProb = p.SnProb
+        self.HhProb = p.HhProb
+
+        # Populate empty probabilities with '9's
+        for pi in range(len(self.BD)):
+            if len(self.BdProb[pi]) < len(self.BD[pi]):
+                self.BdProb[pi] = "9" * len(self.BD[pi])
```

Comment:
[optional] Here's an idea I had that could add some flexibility to this behavior. Instead of filling in with '9', fill in with whatever value was used. This would make it very easy to make a sequence where every value had the probability associated with '5' for example.


A second idea building off of the first. You could fill in the _remaining_ steps using the last value. So for example, in a 8 step sequence the user could specify '95' and it would expand to '95555555'.

Reply:
What a great idea! I can see me using this shorthand all the time! I have added this and updated the doc as per the previous suggestion in my latest commit.

Reply:
These changes look good. It think this is ready to merge now, right?

Reply:
yes please!